WordPress/gutenberg
# Update the position of the patterns tab in the inserter menu

## What?


In the block inserter menu, which can be opened from the "Toggle block inserter" button in the toolbar in the editors,
blocks, patterns and media are displayed as tabs. The order of the tabs were different on different screens; the first tab would show blocks on some screens, and patterns when editing templates.

This PR aims to change the order so that the order is the same in the editors:
1 ) Blocks 2) Patterns 3) Media.

Closes https://github.com/WordPress/gutenberg/issues/47113

## Why?

The inconsistent order may be confusing. Please see more details in the issue.

## How?

The PR removes the "prioritizePatterns" prop and the condition that was used to place the patterns tab before the blocks.
The condition is no longer needed since the order of the tabs should be the same.

## Testing Instructions

**Classic theme:**
Please open the block inserter menu and confirm the order of the tabs on the following screens:
WordPress admin > Posts /Pages > Add new
WordPress admin > Widgets: Confirm that this screen is unchanged and does not show patterns in the tab.
Create or select a user-created block pattern. Test the inserter menu on the pattern creation screen.

**After activating a block theme:**
Please open the block inserter menu and confirm the order of the tabs on the following screens:

1. WordPress admin > Posts /Pages > Add new
2. WordPress admin > Appearance > Editor  > Pages > Open any page which is not also a template
3. WordPress admin > Appearance > Editor > Templates > Open any template
4. WordPress admin > Appearance > Editor + **Click on the main preview window** Select a template part. Select edit in the template part toolbar, to open the "template part editor".
5. WordPress admin > Appearance > Editor > Patterns: Create or select a user-created pattern and test the pattern creation screen.

### Testing Instructions for Keyboard
With a block theme active, create a new post or page.
Inside the Editor top bar region, navigate past the "View posts" link.
Inside the document tools toolbar, activate the button with the aria-label "Toggle block inserter" (The plus icon).
Navigate past the search input field.

With a screen reader, 
"Blocks" should be announced as tab 1 of 3.  Use the right arrow key to select the next tab.
"Patterns" should be announced as tab 2 of 3. Use the right arrow key to select the next tab.
"Media" should tab 3 of 3.

Next please do the same test when editing a block template.
Go to WordPress admin, open the Appearance menu, select Editor, and next, select Templates.
Chose any template in the list, press the Enter key to open that template in the editor.
With the template open, navigate to the tabs in the block inserter menu again and confirm that the tab order is the same:
Blocks, patterns media.
